optimizely/client/client.go

```diff
@@ -364,43 +360,107 @@ func (o *OptimizelyClient) GetAllFeatureVariables(featureKey string, userContext
 	return enabled, variableMap, err
 }
 
-func (o *OptimizelyClient) getFeatureDecision(featureKey string, userContext entities.UserContext) (decisionContext decision.FeatureDecisionContext, featureDecision decision.FeatureDecision, err error) {
+// GetVariation returns the key of the variation the user is bucketed into
+func (o *OptimizelyClient) GetVariation(experimentKey string, userContext entities.UserContext) (result string, err error) {
+
+	defer func() {
+		if r := recover(); r != nil {
+			switch t := r.(type) {
+			case error:
+				err = t
+			case string:
+				err = errors.New(t)
+			default:
+				err = errors.New("unexpected error")
+			}
+			errorMessage := fmt.Sprintf("optimizely SDK is panicking with the error:")
+			logger.Error(errorMessage, err)
+			logger.Debug(string(debug.Stack()))
+		}
+	}()
+
+	_, experimentDecision, err := o.getExperimentDecision(experimentKey, userContext)
+	if err != nil {
+		logger.Error("received an error while computing experiment decision", err)
+	}
+
+	if experimentDecision.Variation != nil {
```

Comment:
same as previous

Reply:
Yeah same response. You can have both nil variation and nil error